fix: clipboard button

llorentelemmc opened the pull request.

mburri commented on AMW_angular/io/src/app/deployments/deployments.component.html:
```diff
@@ -12,6 +12,11 @@
       [message]="errorMessage"
       [messageType]="'warning'"
     ></app-notification>
+    <div *ngIf="showToast" class="toast position-absolute top-0 end-0" role="alert" aria-live="assertive" aria-atomic="true">
```
This should probably go into a simple reusable component.

mburri commented on AMW_angular/io/src/app/deployments/deployments.component.ts:
```diff
@@ -1,5 +1,5 @@
 import { Location } from '@angular/common';
-import { Component, OnInit, ViewChild } from '@angular/core';
+import {ChangeDetectorRef, Component, OnInit, ViewChild} from '@angular/core';
```
ChangeDetectorRef seems to be unused and should be removed

mburri commented on AMW_angular/io/src/app/deployments/deployments.component.ts:
```diff
@@ -22,6 +22,13 @@ declare var $: any;
 @Component({
   selector: 'amw-deployments',
   templateUrl: './deployments.component.html',
+  styles: [`
```
If you create a dedicated toast component, this should go into this component too.

mburri commented on AMW_angular/io/src/app/deployments/deployments.component.ts:
```diff
@@ -289,22 +296,22 @@ export class DeploymentsComponent implements OnInit {
   }
 
-  copyURL() {
+  showToast: boolean = false;
```
Like in java classes, property are usually declared at the top and not in between methods. Just move it up to the top of the class

mburri commented on AMW_angular/io/src/app/deployments/deployments.component.ts:
```diff
@@ -289,22 +296,22 @@ export class DeploymentsComponent implements OnInit {
-      document.execCommand('copy');
-    } catch (e) {
+      await navigator.clipboard.writeText(url);
+      console.log('URL copied to clipboard');
```
With the new toast and the window prompt if something fails, there is no need to log to the console.

mburri commented on AMW_angular/io/src/app/deployments/deployments.component.ts:
```diff
@@ -289,22 +296,22 @@ export class DeploymentsComponent implements OnInit {
+      }, 3000);
+
+    } catch (err) {
+      console.error('Failed to copy: ', err);
```
this console.log can be removed too.

mburri commented:
Seems good to me. @yvespp What do you think of the new Toast component that notifies the user that the URL was successfully copied to the clipboard?

yvespp commented:
Tested it and works. It's not very visible maybe it can be styled similar to the toast of the JSF app?
Example when a resource relation is removed:
![image](https://github.com/liimaorg/liima/assets/15231595/46e29f43-3764-4ca6-b8d9-870824f98ff3)

llorentelemmc commented:
> Tested it and works. It's not very visible maybe it can be styled similar to the toast of the JSF app? Example when a resource relation is removed: ![image](https://user-images.githubusercontent.com/15231595/280722468-46e29f43-3764-4ca6-b8d9-870824f98ff3.png)

![Screenshot from 2023-11-06 17-06-08](https://github.com/liimaorg/liima/assets/71637828/1fca05cf-78f6-489d-a422-cb628d08b839)
@yvespp I've tried to style it more similarly to the notification of the JSF App, is it better like this?

yvespp commented:
@llorentelemmc would it be possible to add the blue part with the exclamation mark? If it's to complicated it's ok like this.

llorentelemmc commented:
@yvespp I've added the blue part with the exclamation part and adjusted the colours a bit. What do you think?
![Screenshot from 2023-11-09 17-20-23](https://github.com/liimaorg/liima/assets/71637828/5f3e61f0-ee5d-41b0-90b6-296937e0e895)

yvespp commented:
Cool thanks!
